src/applications/vaos/tests/new-appointment/components/DateTimeRequestPage.unit.spec.jsx

```diff
@@ -1,213 +1,364 @@
+      // 1. Simulate user selecting a date
+      userEvent.click(buttons[0]);
+
+      // 2. Simulate user selecting AM
+      let checkbox = screen.getByRole('checkbox', {
+        name: 'AM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 3. Simulate user selecting another date
+      userEvent.click(buttons[1]);
+
+      // 3. Simulate user selecting PM
+      checkbox = screen.getByRole('checkbox', {
+        name: 'PM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 4. Simulate user selecting another date
+      userEvent.click(buttons[2]);
+
+      // 5. Simulate user selecting AM
+      checkbox = screen.getByRole('checkbox', {
+        name: 'AM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 6. Simulate user selecting another date
+      userEvent.click(buttons[3]);
+
+      // 7. Simulate user selecting PM which should result in error
+      checkbox = screen.getByRole('checkbox', {
+        name: 'PM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // NOTE: alert doesn't have a name so search for text too
+      expect(screen.getByRole('alert')).to.be.ok;
+      expect(
+        screen.getByText(
+          'You can only choose up to 3 dates for your appointment.',
+        ),
+      ).to.be.ok;
+    });
+
+    it('should display an alert when user selects 2 dates and multiple times', async () => {
+      const store = createTestStore({
+        newAppointment: {
+          data: {
+            calendarData: {},
+          },
+          pages: [],
+          eligibility: [],
+          appointmentSlotsStatus: FETCH_STATUS.succeeded,
+        },
+      });
+
+      const screen = renderWithStoreAndRouter(
+        <Route component={DateTimeRequestPage} />,
+        {
+          store,
+        },
+      );
+
+      // Find all available appointments for the current month
+      const currentMonth = moment().format('MMMM');
+      const buttons = screen
+        .getAllByRole('button', { name: new RegExp(`${currentMonth}`) })
+        .filter(button => button.disabled === false);
+
+      // it should display an alert when the users selects more than the allowed dates
+      // 1. Simulate user selecting a date
+      userEvent.click(buttons[0]);
+
+      // 2. Simulate user selecting AM
+      let checkbox = screen.getByRole('checkbox', {
+        name: 'AM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 3. Simulate user selecting PM
+      checkbox = screen.getByRole('checkbox', {
+        name: 'PM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 4. Simulate user selecting another date
+      userEvent.click(buttons[1]);
+
+      // 5. Simulate user selecting AM
+      checkbox = screen.getByRole('checkbox', {
+        name: 'AM appointment',
+      });
+      userEvent.click(checkbox);
+
+      // 6. Simulate user selecting PM which should result in error
+      checkbox = screen.getByRole('checkbox', {
```

Comment:
I think we want to have an await here when picking the checkbox to click.

I had to do that here for a similar test: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/tests/new-appointment/components/DateTimeSelectPage/select-date-page.unit.spec.jsx#L146